lib/indico.rb

```diff
@@ -137,16 +137,17 @@ def self.collections(config = nil)
 
   class Collection
 
-      def initialize(collection)
+      def initialize(collection, config = nil)
         if collection.kind_of?(String)
-          @collection = collection
-        else
-          raise TypeError, "Collection must be initialized with a String name"
+          @collection = collection
+          @domain = config.nil? ? nil : config["domain"]

```

Comment:
Could this raise the ruby equivalent of a key error here?  What happens if config["domain"] doesn't exist?